integrations/Matomo/browser.js

```diff
@@ -0,0 +1,195 @@
+    this.enableLinkTracking = config.enableLinkTracking;
+    this.disablePerformanceTracking = config.disablePerformanceTracking;
+    this.enableCrossDomainLinking = config.enableCrossDomainLinking;
+    this.setCrossDomainLinkingTimeout = config.setCrossDomainLinkingTimeout;
+    this.timeout = config.timeout;
+    this.getCrossDomainLinkingUrlParameter =
+      config.getCrossDomainLinkingUrlParameter;
+    this.disableBrowserFeatureDetection = config.disableBrowserFeatureDetection;
+
+    this.ecomEvents = {
+      SET_ECOMMERCE_VIEW: "SET_ECOMMERCE_VIEW",
+      ADD_ECOMMERCE_ITEM: "ADD_ECOMMERCE_ITEM",
+      REMOVE_ECOMMERCE_ITEM: "REMOVE_ECOMMERCE_ITEM",
+      TRACK_ECOMMERCE_ORDER: "TRACK_ECOMMERCE_ORDER",
+    };
+  }
+
+  loadScript() {
+    const _paq = (window._paq = window._paq || []);
+    (function (serverUrl, siteId) {
+      let u = serverUrl;
+      window._paq.push(["setTrackerUrl", `${u}matomo.php`]);
+      window._paq.push(["setSiteId", siteId]);
+      const d = document;
+      const g = d.createElement("script");
+      const s = d.getElementsByTagName("script")[0];
+      g.async = true;
+      u = u.replace("https://", "");
+      g.src = `//cdn.matomo.cloud/${u}matomo.js`;
+      s.parentNode.insertBefore(g, s);
+    })(this.serverUrl, this.siteId);
+  }
+
+  init() {
+    logger.debug("===In init Matomo===");
+    this.loadScript();
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Matomo===");
+    return !!(window._paq && window._paq.push !== Array.prototype.push);
+  }
+
+  isReady() {
+    logger.debug("===In isReady Matomo===");
+
+    // Dasboard Event Settings
+    if (window._paq && window._paq.push !== Array.prototype.push) {
+      if (this.trackAllContentImpressions) {
+        window._paq.push(["trackAllContentImpressions"]);
+      }
+      if (
+        this.trackVisibleContentImpressions &&
+        this.checkOnScroll &&
+        this.timeIntervalInMs
+      ) {
+        window._paq.push([
+          "trackVisibleContentImpressions",
+          this.checkOnScroll,
+          this.timeIntervalInMs,
+        ]);
+      }
+      if (this.logAllContentBlocksOnPage) {
+        window._paq.push(["logAllContentBlocksOnPage"]);
+      }
+      if (this.enableHeartBeatTimer && this.activeTimeInseconds) {
+        window._paq.push(["enableHeartBeatTimer", this.activeTimeInseconds]);
+      }
+      if (this.enableLinkTracking) {
+        window._paq.push(["enableLinkTracking", true]);
+      }
+      if (this.disablePerformanceTracking) {
+        window._paq.push(["disablePerformanceTracking"]);
+      }
+      if (this.enableCrossDomainLinking) {
+        window._paq.push(["enableCrossDomainLinking"]);
+      }
+      if (this.setCrossDomainLinkingTimeout && this.timeout) {
+        window._paq.push(["setCrossDomainLinkingTimeout", this.timeout]);
+      }
+      if (this.getCrossDomainLinkingUrlParameter) {
+        window._paq.push(["getCrossDomainLinkingUrlParameter"]);
```

Comment:
Would it better if we give some info about these fields here so that we dont have to check matomo docs everytime easier for next developer to work on this

Reply:
Added